a8c-files.php

```diff
@@ -776,6 +776,232 @@ function a8c_files_init() {
 	new A8C_Files();
 }
 
+class A8C_Files_Image_Srcset_Meta {
+
+	/**
+	 * Create proper 'sizes' array for the image meta including the Photon query args used on the img src.
+	 *
+	 * @param array  $image_meta    The image meta data as returned by 'wp_get_attachment_metadata()'.
+	 * @param array  $size_array    Array of width and height values in pixels (in that order).
+	 * @param string $image_src     The 'src' of the image.
+	 * @param int    $attachment_id The image attachment ID or 0 if not supplied.
+	 * @return array                The normalized image meta data or empty $image_meta['sizes'] on error.
+	 */
+	public static function generate_image_meta( $image_meta, $size_array, $image_src, $attachment_id ) {
+		// Reset $image_meta['sizes'] if not empty. Need to build it again matching the current img src.
+		$image_meta['sizes'] = array();
+
+		// Bail out early if the image is not hosted on the site, missing image meta or original image is too small.
+		// This will disable the generation of srcset.
+		if ( strpos( $image_src, home_url() ) === false ||
```

Comment:
This is rather fragile, and needs to be rethought. For starters, we should be comparing against just the hosts, using `parse_url()`. Otherwise, `https://www.example.com` and `https://www.example.com.bad.site` would pass, while a site with inconsistent HTTPS won't be parsed. Additionally, this doesn't account for differences between `siteurl` and `home`, and we have some sites with split settings. Either a filter, or checking both `home_url()` and `site_url()`, is needed.